### ISSUE TITLE
pcre2

### ISSUE THREAD

**Issue by rwmitchell** [ascii-boxes/boxes#94]
Is needing libpcre2 a new requirement?  I last compiled boxes on 2020-01-14 and do not remember any issues.  Just updated and now the compile is failing on not finding pcre2.h.

Previously compiled on a couple versions back of OS X, have since updated to Big Sur.

**Comment by tsjensen** [comment]
Right, since v2.0.0, pcre2 is a requirement.  
You can find detailed [build instructions](https://boxes.thomasjensen.com/build.html) on the website.

**Comment by tsjensen** [comment]
Since you're on Mac, you might also be interested to install via *brew*, or look at their [boxes formula](https://github.com/Homebrew/homebrew-core/blob/eff4fcb480fa75ba8b9a58e67138373928db3298/Formula/boxes.rb) for additional inspiration on how to compile yourself.

**Comment by rwmitchell** [comment]
I don't use brew.  If the source doesn't compile, I don't use it.  Even with libpcre2, this doesn't

LDFLAGS and CFLAGS are ignored by the Makefile:

printf "%s\n%s\n" $CFLAGS $LDFLAGS
-I/Users/USER/local/include
-L/Users/USER/local/lib

(zsh, both are exported)

make used ```gcc -I. -I../src -Wall -W -O    -c -o parser.o parser.c````

After changing src/Makefile to use the environment CFLAGS, it still generated errors in parser.y.

For a 'just for fun' program, this isn't worth it.   Thanks anyway.

**Comment by tsjensen** [comment]
It appears you are setting `CFLAGS` and `LDFLAGS` in your shell before calling make. This wouldn't work because they are set in the Makefile, too. We have `CFLAGS_ADDTL` and `LDFLAGS_ADDTL` for adding arguments to compiler and linker calls. (I don't have access to a Mac to try and reproduce your problem.)

**Comment by rwmitchell** [comment]
If your Makefiles used: ```$(eval CFLAGS := $(CFLAGS) -I. -Iregexp -Wall -W $(CFLAGS_ADDTL))``` then it would use the user supplied value in addition to yours.  There is probably even a better way.

You can't possibly know the include path a user has used for a separately installed library.  Requiring them to edit the Makefile is poor method to fix it.

**Comment by tsjensen** [comment]
> You can't possibly know the include path a user has used for a separately installed library. Requiring them to edit the Makefile is poor method to fix it.

It's actually not required to edit the Makefile. You can type something like this:

    CFLAGS_ADDTL=-I/Users/USER/local/include LDFLAGS_ADDTL=-L/Users/USER/local/lib make

Let me know if stuff works for you now. We seem to have drifted somewhat from the original topic of runtime dependencies.

**Comment by rwmitchell** [comment]
Neither of those are standard nor listed in any of the readme files ( that I searched anyway ).  If it isn't documented and you're doing something non-standard, you can't expect people to set it properly or know they need to set it.

**Comment by tsjensen** [comment]
> Neither of those are standard nor listed in any of the readme files

Good point! I added that information to the [FAQ](https://boxes.thomasjensen.com/faq.html#q5).
